ext/RMagick/rmimage.c

```diff
@@ -4104,14 +4480,68 @@ Image_morphology_channel(VALUE self, VALUE channel_v, VALUE method_v, VALUE iter
 
     exception = AcquireExceptionInfo();
 
+#if defined(IMAGEMAGICK_7)
+    BEGIN_CHANNEL_MASK(image, channel);
+    new_image = MorphologyImage(image, method, NUM2LONG(iterations), kernel, exception);
+    CHANGE_RESULT_CHANNEL_MASK(new_image);
+    END_CHANNEL_MASK(image);
```

Comment:
That would not really be an option here because the signature of the methods inside that block would be very different and you would need a method for every signature.